### ISSUE TITLE
For SENet152 model, ORT1.10.0 runs much slower than ORT1.7.0, only 1/8

### ISSUE THREAD

**Issue by yetingqiaqia** [microsoft/onnxruntime#10433]
**Describe the bug**
Hi ort team,

We have many models running on ORT. Thanks for previous bug fix on https://github.com/microsoft/onnxruntime/issues/9162, we are able to use ORT1.10.0 run ResNet on ORT with similar speed as pyTorch.

However, when testing ORT1.10.0 on the throughput of other models, I notice there is an obvious throughput drop on SEN152 model -- ORT1.10.0 took 8x inference time of ORT1.7.0.

| ORT version| Warm-up (s) | Inference Time for 10 batches (s) |
| ------------- | ------------- | ------------- |
| ORT1.10.0 | 7.90 | **47.80** |
| ORT1.7.0 | 3.67 | **6.45** |

Note: for ORT1.10.0, I also tested other speed optimization methods like cudnn_conv_use_max_workspace='1', IOBinding. There is no obvious speed difference. To keep this ticket simple, I didn't add those tests in below example code.

**Urgency**
We are running an ONNX inference platform for Microsoft ads users. This is a blocking issue for us to do docker image update.

**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Ubuntu 18.04.2 LTS
- ONNX Runtime installed from (source or binary): binary
- ONNX Runtime version: 1.7.0 and 1.10.0, all installed by pip command
- - 1.10.0: pip install onnxruntime-gpu==1.10.0
- - 1.7.0: pip uninstall onnxruntime-gpu
     pip install onnxruntime-gpu==1.7.0
- Python version: 3.6.9
- Visual Studio version (if applicable): N/A
- GCC/Compiler version (if compiling from source): N/A
- CUDA/cuDNN version: Driver Version: 495.29.05 CUDA Version: 11.5
- - I also tested a lower version: Driver Version: 455.45.01 CUDA Version: 11.1. The speed issue still exists. So, it doesn't like to relate with CUDA. 
- GPU model and memory: V100, 16GB

**To Reproduce**
- Describe steps/code to reproduce the behavior.

The speed issue should be able to reproduce by simply running below code in ORT1.10.0 and ORT1.7.0 environment:

```python
import numpy as np
import time
np.random.seed(123)

def test():
    #Load ort model
    import onnxruntime as ort
    sess_options = ort.SessionOptions()
    sess_options.graph_optimization_level = ort.GraphOptimizationLevel.ORT_ENABLE_ALL
    sess_options.intra_op_num_threads = 0
    onnx_model_path = "./senet152.onnx"
    sess = ort.InferenceSession(onnx_model_path, sess_options, providers=['CUDAExecutionProvider'])
    input_name = sess.get_inputs()[0].name
    label_name = sess.get_outputs()[0].name

    data = np.random.rand(128, 3, 224, 224).astype(np.float32)

    #warm-up run
    warm_up_start_stamp = time.time()
    onnx_outs = sess.run([label_name], {input_name: data})[0]
    print(f"It takes {time.time()-warm_up_start_stamp} to finish warm-up.\n")

    start_stamp = time.time()
    num_batches = 10
    for i in range(num_batches):
        print(f"batch id: {i}")
        onnx_outs = sess.run([label_name], {input_name: data})[0]
        #print(f"onnx_outs:", onnx_outs)
        #np.testing.assert_allclose(torch_outs, onnx_outs, rtol=1e-03, atol=1e-05)
        print(f"{i}th batch finished successfully. ")
    print(f"It takes {time.time()-start_stamp} to finish {num_batches} batches.\n")

test()
```

- Attach the ONNX model to the issue (where applicable) to expedite investigation.
https://drive.google.com/file/d/1chojjAKFavdGXReGyZGblLCeT_TIbHkP/view?usp=sharing

**Expected behavior**
I expect the ORT1.10.0 inference speed for SENet152 model should be at least the same as ORT1.7.0

**Comment by hariharans29** [comment]
If you can post the nvprof logs from the 2 ORT versions, it may provide some clue. But possibly there are multiple issues if the latency is 8x larger.

**Comment by yetingqiaqia** [comment]
I did more tests, the regression started in 1.8.0. See below table, hope this could help narrow down your investigation.
| ORT version| Warm-up (s) | Inference Time for 10 batches (s) |
| ------------- | ------------- | ------------- |
| ORT1.7.0 | 3.67 | **6.45** |
| ORT1.8.0 | 14.44 | **47.79** |
| ORT1.8.1 | 7.90 | **47.79** |
| ORT1.9.0 | 7.91 | **47.81** |
| ORT1.10.0 | 7.90 | **47.80** |

**Comment by yetingqiaqia** [comment]
I did more tests, with newer CUDA versions. This speed issue disappeared in CUDA11.6 (Nvidia docker nvcr.io/nvidia/tensorflow:22.01-tf1-py3). So, I feel we could close this ticket now. Thanks.

**Comment by hariharans29** [comment]
Thanks for letting us know.

As you reported, I did get similar drop in perf with ORT 1.10 with CUDA 11.1 when compared with ORT 1.7 with CUDA 11.1 (I didn't switch to the CUDA version ORT 1.7 was built with).  I was able to get back the ORT 1.7 perf with ORT 1.10 by setting the `cudnn_conv_algo_search` to `DEFAULT` (`EXHAUSTIVE` is the out-of-the-box option) on my Win 10 + K620 card machine. So, it seems like there is some variance between CUDA/CuDNN versions as to which conv algo gets picked for the same filter shapes/ input shapes and that may be causing the discrepancy.
